pkg/volume/volume.go

```diff
@@ -132,22 +132,22 @@ type Deleter interface {
 // Attacher can attach a volume to a node.
 type Attacher interface {
 	// Attach the volume specified by the given spec to the given host
-	Attach(spec *Spec, hostName string) error
+	Attach(hostName string, mounter mount.Interface) error
```

Comment:
The mount interface should not be passed to the `Attach` method. Attach must not be aware of any mounting specific logic or semantics since it will be called by the attach/detach controller running on a different physical machine.